Enforce typechecks

I noticed you use typehints

```
    def parse_file(self, file_content: str) -> None:
```

And you have `mypy` installed an indirect dependency.

But your docs and CI doesn't run mypy.

So you might want to add that to enforce types locally and on CI. Or maybe you are fine with how it is now.

---

Here is the output I got locally. These may be bugs in the code, or types set incorrectly or just things Mypy is strict about (and you may not care about addressing these issues).

```sh
mypy -p make_to_batch
```
```
make_to_batch/makefile.py:211: error: Value of type "Collection[str]" is not indexable
make_to_batch/makefile.py:212: error: Unsupported left operand type for + ("Collection[str]")
make_to_batch/makefile.py:219: error: Incompatible types in assignment (expression has type "str", variable has type "List[str]")
make_to_batch/makefile.py:220: error: No overload variant of "sub" matches argument types "str", "str", "List[str]"
make_to_batch/makefile.py:220: note: Possible overload variants:
make_to_batch/makefile.py:220: note:     def [AnyStr in (str, bytes)] sub(pattern: AnyStr, repl: AnyStr, string: AnyStr, count: int = ..., flags: Union[int, RegexFlag] = ...) -> AnyStr
make_to_batch/makefile.py:220: note:     def [AnyStr in (str, bytes)] sub(pattern: AnyStr, repl: Callable[[Match[AnyStr]], AnyStr], string: AnyStr, count: int = ..., flags: Union[int, RegexFlag] = ...) -> AnyStr
make_to_batch/makefile.py:220: note:     <2 more similar overloads not shown, out of 4 total overloads>
make_to_batch/makefile.py:221: error: Incompatible return value type (got "List[str]", expected "str")
make_to_batch/cli.py:26: error: Cannot find implementation or library stub for module named 'colorama'
make_to_batch/cli.py:26: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 6 errors in 2 files (checked 6 source files)
```

--- a/.flake8
+++ b/.flake8
@@ -1,0 +1,2 @@
+[flake8]
+exclude = .git,__pycache__,docs,tests

--- a/.github/workflows/pythonpackage.yml
+++ b/.github/workflows/pythonpackage.yml
@@ -34,6 +34,7 @@ jobs:
         pipenv run flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         pipenv run flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+        pipenv run mypy -p make_to_batch
     - name: Test with pytest
       run: |
         pipenv run pip install .

--- a/.github/workflows/pythonpackage.yml
+++ b/.github/workflows/pythonpackage.yml
@@ -34,6 +34,8 @@ jobs:
         pipenv run flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         pipenv run flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Lint with mypy
+      run: |
         pipenv run mypy -p make_to_batch
     - name: Test with pytest
       run: |

--- a/Pipfile
+++ b/Pipfile
@@ -7,9 +7,11 @@ verify_ssl = true
 pdoc3 = "*"
 flake8 = "*"
 pytest = "*"
+mypy = "*"
 
 [packages]
 colorama = "*"
 importlib-metadata = "==1.7.0"
+typing_extensions = "*"
 
 [requires]

--- a/Pipfile
+++ b/Pipfile
@@ -4,10 +4,10 @@ url = "https://pypi.org/simple"
 verify_ssl = true
 
 [dev-packages]
-pdoc3 = "*"
 flake8 = "*"
 pytest = "*"
 mypy = "*"
+types-colorama = "*"
 
 [packages]
 colorama = "*"

--- a/make_to_batch/look_up_table.py
+++ b/make_to_batch/look_up_table.py
@@ -39,6 +39,10 @@ keys:
 * ``options``: a dictionary containing, for each option of the Unix command,
   the corresponding option in Windows.
 """
+from typing import Dict
+from typing_extensions import TypedDict
+
+LookUpTableContent = TypedDict('LookUpTableContent', {'command': str, 'options': Dict[str, str]})
 
 linux_to_dos = {
     'mkdir': {
@@ -70,4 +74,4 @@ linux_to_dos = {
             '--help': '/?',
         },
     },
-}
+}  # type: Dict[str, LookUpTableContent]

--- a/make_to_batch/makefile.py
+++ b/make_to_batch/makefile.py
@@ -216,9 +216,9 @@ class Makefile:
         for _ in range(number_of_dir_changed):
             batch_commands.append("POPD")
 
-        batch_commands = re.sub(r"\$[({](.*?)[)}]", r"%\1%", " && ".join(batch_commands))
-        batch_commands = re.sub(r"%MAKE%", r"CALL make.bat", batch_commands)
-        return batch_commands
+        batch_commands_str = re.sub(r"\$[({](.*?)[)}]", r"%\1%", " && ".join(batch_commands))
+        batch_commands_str = re.sub(r"%MAKE%", r"CALL make.bat", batch_commands_str)
+        return batch_commands_str
 
     def to_batch(self) -> str:
         """Convert the Makefile to a Batch file.

